llvm/lib/Object/GOFFObjectFile.cpp

```diff
@@ -481,10 +477,7 @@ GOFFObjectFile::getSectionContents(DataRefImpl Sec) const {
     ESDRecord::getFillByteValue(EdEsdRecord, FillByte);
 
   // Initialize section with fill byte.
-  auto DataPtr =
-      std::make_pair(SectionSize, std::make_unique<uint8_t[]>(SectionSize));
-  uint8_t *Data = DataPtr.second.get();
-  memset(Data, FillByte, SectionSize);
+  SmallVector<uint8_t> Data(SectionSize, FillByte);
```

Comment:
Elsewhere, you've used the version with `0` as the second template parameter, whereas here you've omitted it. Could you explain why you've decided to use each version (in review comments, not code comments)?